ui/utils.py

```diff
@@ -16,3 +16,22 @@ def wrapper(self, *args, **kwargs):
 
 def clamp(value, _min, _max):
     return max(_min, min(value, _max))
+
+
+def check_value_lock(func):
+    # A safety check wrapper so that there's no race conditions between functions able to change position/value
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        assert self.value_lock, "Class has no member self.value_lock"  # todo:maybe we should create it here ?
+        # Value-changing code likely to run in concurrent thread and therefore we need a lock
+        # if self.__locked_name__ is not None: print("Another function already locked the thread! Name is {}, current is {}".format(self.__locked_name__, func.__name__))
+        self.value_lock.acquire()
+        # self.__locked_name__ = func.__name__
+        # print("Locked function {}".format(func.__name__))
+        result = func(self, *args, **kwargs)
+        self.value_lock.release()
+        # print("Unlocked function {}".format(func.__name__))
```

Comment:
why not using the naming thing that was there earlier ?

Reply:
Which naming thing do you mean?

Reply:
I'm talking about the commented print functions. I think it's a very good idea but I don't know why they were commented out. Maybe raising an exception with the name would be good as well.

Reply:
They were left over from debugging, and they might be useful again. If you could convert them into `logger.debug()` calls, that'd be great!